src/components/PopupWithForm.js

```diff
@@ -1,37 +1,46 @@
-import Popup from './Popup.js';
-
+import Popup from "./Popup.js";
 class PopupWithForm extends Popup {
-    constructor(popupSelector, handleFormSubmit) {
-        super({ popupSelector }); // { popupSelector: "#edit-modal"}
-        this._form = this._popupElement.querySelector('.modal__form');
-        this._popupForm = this._popupElement.querySelector('.modal__form');
-        this._handleFormSubmit = handleFormSubmit;
-        }
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._form = this._popupElement.querySelector(".modal__form");
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  _getInputValues() {
+    this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._formValues = {};
+    this._inputList.forEach(
+      (input) => (this._formValues[input.name] = input.value)
+    );
+    return this._formValues;
+  }
 
-        _getInputValues() {
-            this._inputList = this._popupElement.querySelectorAll(".modal__input");
-            this._formValues = {};
-            this._inputList.forEach(
-                (input) => (this._formValues[input.name] = input.value)
-            );
-            return this._formValues
-        }
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupElement.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit(this._getInputValues());
+    });
+  }
 
-        setEventListeners() {
-            this._popupElement.addEventListener("submit", (evt) => {
-                this._handleFormSubmit(this._getInputValues());
-                evt.preventDefault();
-            });
-        }
+  open() {
+    super.open();
```

Comment:
* You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.